File: bootstrap/archived_runa_0.0.7.5+/v0.0.8.4.5.2/benchmarks/java/Nqueens.java
public class Nqueens {
    static boolean isSafe(int[][] board, int row, int col, int n) {
        for (int i = 0; i < col; i++) {
            if (board[row][i] == 1) return false;
        }

        for (int i = row - 1, j = col - 1; i >= 0 && j >= 0; i--, j--) {
            if (board[i][j] == 1) return false;
        }

        for (int i = row + 1, j = col - 1; i < n && j >= 0; i++, j--) {
            if (board[i][j] == 1) return false;
        }

        return true;
    }

    static boolean solve(int[][] board, int col, int n) {
        if (col >= n) return true;

        for (int i = 0; i < n; i++) {
            if (isSafe(board, i, col, n)) {
                board[i][col] = 1;
                if (solve(board, col + 1, n)) return true;
                board[i][col] = 0;
            }
        }

        return false;
    }

    public static void main(String[] args) {
        int n = 8;
        int[][] board = new int[n][n];
        solve(board, 0, n);
        System.exit(0);
    }
}

File: bootstrap/v0.0.8.4.5.2.5/benchmarks/python/nqueens.py
def is_safe(board, row, col, n):
    for i in range(col):
        if board[row][i]:
            return False

    i, j = row - 1, col - 1
    while i >= 0 and j >= 0:
        if board[i][j]:
            return False
        i -= 1
        j -= 1

    i, j = row + 1, col - 1
    while i < n and j >= 0:
        if board[i][j]:
            return False
        i += 1
        j -= 1

    return True

def solve(board, col, n):
    if col >= n:
        return True

    for i in range(n):
        if is_safe(board, i, col, n):
            board[i][col] = 1
            if solve(board, col + 1, n):
                return True
            board[i][col] = 0

    return False

def main():
    n = 8
    board = [[0 for _ in range(n)] for _ in range(n)]
    solve(board, 0, n)
    return 0

if __name__ == "__main__":
    exit(main())
